### PR TITLE
fix: pass undefined to test:before:after:run:async if there isn't a next test

### PR DESCRIPTION
**Pull request by mschile**
### Additional details

Updated the `test:before:after:run:async` event to not pass `nextTestHasTestIsolationOn` if there isn't a next test (i.e. it's the last test).

### PR Tasks

- [x] Have tests been added/updated?
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Review by ryanthemanuel — APPROVED** [review]

File: cli/CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@ _Released 11/7/2023 (PENDING)_
 **Bugfixes:**
 
 - Fixed an issue determining visibility when an element is hidden by an ancestor with a shared edge. Fixes [#27514](https://github.com/cypress-io/cypress/issues/27514).
+- Fixed an issue where `nextTestHasTestIsolationOn` was incorrectly being set when there isn't a next test. Addressed in [#28213](https://github.com/cypress-io/cypress/pull/28213)
```
Comment (chrisbreiding) [inline cli/CHANGELOG.md]: `test:before:after:run:async` and `nextTestHasTestIsolationOn` aren't publicly documented, so I'm not sure this is meaningful for an end user. Is there an effect this has or a bug this fixes that affects users?

Comment (mschile) [reply cli/CHANGELOG.md]: Good point. It _may_ help some protocol users where CDP events were still being processed at the end of the spec.

Comment (mschile) [reply cli/CHANGELOG.md]: Maybe I should update the text to be something like:
"Stopped processing CDP events at the end of a spec when Test Isolation is off and Test Replay is enabled."

**Review by cacieprins — APPROVED** [review]

File: cli/CHANGELOG.md (hunk 1 above)
Comment (chrisbreiding) [reply cli/CHANGELOG.md]: That sounds good

Comment (mschile) [reply cli/CHANGELOG.md]: updated [08acec7](https://github.com/cypress-io/cypress/pull/28213/commits/08acec70ce3450238a93d4ed14642b794506e54e)

**Review by chrisbreiding — APPROVED** [review]

**Pull request merged**